pubsub/subscriber.go

```diff
@@ -0,0 +1,96 @@
+package pubsub
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"encoding/json"
+	"errors"
+	queuesgo "github.com/merlinapp/queues-go"
+	"log"
+	"strconv"
+)
+
+type subscriber struct {
+	project          string
+	subscriptionName string
+	elements         []routerElement
+}
+
+type routerElement struct {
+	event       string
+	handlerFunc queuesgo.HandlerFunc
+}
+
+/*
+Creates a new Google's pubsub subscriber implementation
+the subscription name must exists already on the given projects
+*/
+func NewSubscriber(project, subscriptionName string) queuesgo.Subscriber {
```

Comment:
Should this also have the expected type?

Reply:
Done